contracts/TokenManager.sol

```diff
@@ -189,6 +273,262 @@ contract TokenManager is System, IApplication {
+    MirrorAckPackage memory mirrorAckPackage;
+    RLPDecode.Iterator memory iter = msgBytes.toRLPItem().iterator();
+    bool success = false;
+    uint256 idx=0;
+    while (iter.hasNext()) {
+      if (idx == 0)      mirrorAckPackage.mirrorSender   = iter.next().toAddress();
+      else if (idx == 1) mirrorAckPackage.bep20Addr      = iter.next().toAddress();
+      else if (idx == 2) mirrorAckPackage.bep20Decimals  = uint8(iter.next().toUint());
+      else if (idx == 3) mirrorAckPackage.bep2Symbol     = bytes32(iter.next().toUint());
+      else if (idx == 4) mirrorAckPackage.mirrorFee      = iter.next().toUint();
+      else if (idx == 5) {
+        mirrorAckPackage.errorCode  = uint8(iter.next().toUint());
+        success = true;
+      }
+      else break;
+      idx++;
+    }
+    return (mirrorAckPackage, success);
+  }
+
+  function mirror(address bep20Addr, uint64 expireTime) payable public returns (bool) {
+    require(ITokenHub(TOKEN_HUB_ADDR).getBep2SymbolByContractAddr(bep20Addr) == bytes32(0x00), "already bound");
+    require(!mirrorPendingRecord[bep20Addr], "mirror pending");
+    uint256 miniRelayFee = ITokenHub(TOKEN_HUB_ADDR).getMiniRelayFee();
+    require(msg.value%TEN_DECIMALS == 0 && msg.value>=mirrorFee.add(miniRelayFee), "msg.value must be N * 1e10 and greater than sum of miniRelayFee and mirrorFee");
+    require(expireTime>=block.timestamp + 120 && expireTime <= block.timestamp + 86400, "expireTime must be two minutes later and one day earlier");
+    uint8 decimals = IBEP20(bep20Addr).decimals();
+    uint256 totalSupply = IBEP20(bep20Addr).totalSupply();
+    require(convertToBep2Amount(totalSupply, decimals) <= MAX_BEP2_TOTAL_SUPPLY, "too large total supply");
+    string memory name = IBEP20(bep20Addr).name();
+    bytes memory nameBytes = bytes(name);
+    require(nameBytes.length>=1 && nameBytes.length<=32, "name length must be in [1,32]");
+    string memory symbol = IBEP20(bep20Addr).symbol();
+    bytes memory symbolBytes = bytes(symbol);
+    require(symbolBytes.length>=MINIMUM_BEP20_SYMBOL_LEN && symbolBytes.length<=MAXIMUM_BEP20_SYMBOL_LEN, "symbol length must be in [3,8]");
+    for (uint8 i = 0; i < symbolBytes.length; i++) {
+      require((symbolBytes[i]>='a' && symbolBytes[i]<='a') || (symbolBytes[i]>='A' && symbolBytes[i]<='Z'), "symbol must not contain non-alphabet");
+    }
+    address(uint160(TOKEN_HUB_ADDR)).transfer(msg.value.sub(mirrorFee));
+    mirrorPendingRecord[bep20Addr] = true;
+    bytes32 bytes32Name;
+    assembly {
+      bytes32Name := mload(add(name, 32))
+    }
+    bytes32 bytes32Symbol;
+    assembly {
+      bytes32Symbol := mload(add(symbol, 32))
+    }
+    MirrorSynPackage memory mirrorSynPackage = MirrorSynPackage({
+      mirrorSender:  msg.sender,
+      bep20Addr:     bep20Addr,
+      bep20Name:     bytes32Name,
+      bep20Symbol:   bytes32Symbol,
+      bep20Supply:   totalSupply,
+      bep20Decimals: decimals,
+      mirrorFee:     mirrorFee/TEN_DECIMALS,
```

Comment:
use safe div please.

Reply:
safe div just ensure the `TEN_DECIMALS ` is not zero. Not necessary here

Reply:
Plesae make the uniform style.  Either change line 371 or here.